gateway/src/apicast/policy/find_service/find_service.lua

```diff
@@ -76,8 +29,18 @@ end
 
 _M.rewrite = find_service
 
--- ssl_certiticate is the first phase executed when request arrives on HTTPS
--- therefore it needs to find a service to build a policy chain
-_M.ssl_certificate = find_service
+-- ssl_certificate is the first phase executed when request arrives on HTTPS
+-- therefore it needs to find a service to build a policy chain.
+-- The method and the path are not available in the ssl_certificate phase, so
+-- path-based routing does not work. It should always find the service by host.
+function _M:ssl_certificate(context)
+  if self.find_service ~= host_based_finder.find_service then
+    ngx.log(ngx.WARN, 'Configured to do path-based routing, but it is not',
+                      'compatible with TLS. Falling back to routing by host.')
+  end
+
+  context.service = context.service or
+                    host_based_finder.find_service(context.configuration, context.host)
```

Comment:
Maybe you can store just `find_service` in the local variable, so the JIT has it easier and does not have to check if the table changed.

But I see that because of the testing code and stubs you actually need it to be in a table. That is unfortunate.